internal/devbox/packages.go

```diff
@@ -290,10 +293,21 @@ func (d *Devbox) ensureStateIsUpToDate(ctx context.Context, mode installMode) er
 		return err
 	}
 
-	// Use the printDevEnvCache if we are adding or removing or updating any package,
-	// AND we are not in the shellenv-enabled environment of the current devbox-project.
-	usePrintDevEnvCache := mode != ensure && !d.IsEnvEnabled()
-	if _, err := d.computeEnv(ctx, usePrintDevEnvCache); err != nil {
```

Comment:
This undos an optimization which is to skip recomputing envs for add/rm when not in a devbox environment. 

I think the reason we introduced that is because sometimes this triggered some slow stuff like downloading nixpkgs for some other package. I'm not entirely sure this is fixable, just a tradeoff.